lib/datadog/lambda.rb

```diff
@@ -58,5 +63,49 @@ def self.metric(name, value, time: nil, **tags)
       metric = { e: time_ms, m: name, t: tag_list, v: value }.to_json
       puts metric
     end
+
+    # Generate tags for enhanced metrics
+    # @param context [Object] https://docs.aws.amazon.com/lambda/latest/dg/ruby-context.html
+    # @return [hash] a hash of the enhanced metrics tags
+    def self.gen_enhanced_tags(context)
+      arn_parts = context.invoked_function_arn.split(':')
+      {
+        functionname: context.function_name,
+        region: arn_parts[3],
+        account_id: arn_parts[4],
+        memorysize: context.memory_limit_in_mb,
+        cold_start: @is_cold_start,
+        runtime: "Ruby #{RUBY_VERSION}"
+      }
+    rescue StandardError => _e
+      {}
```

Comment:
Let's add a debug log for this error. It should help us identify when customer's haven't called wrap correctly eg.
```
Datadog::Utils.logger.debug "couldn't read lambda context obj #{e}"
```